core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestDecorrelateLeftUnnestWithGlobalAggregation.java

```diff
@@ -0,0 +1,390 @@
+                                                        Optional.empty(),
+                                                        LEFT,
+                                                        Optional.empty(),
+                                                        assignUniqueId("unique", values("corr")))))));
+    }
+
+    @Test
+    public void testPreprojectUnnestSymbol()
+    {
+        tester().assertThat(new DecorrelateLeftUnnestWithGlobalAggregation())
+                .on(p -> p.correlatedJoin(
+                        ImmutableList.of(p.symbol("corr")),
+                        p.values(p.symbol("corr")),
+                        p.aggregation(builder -> builder
+                                .globalGrouping()
+                                .addAggregation(p.symbol("max"), PlanBuilder.expression("max(unnested_char)"), ImmutableList.of(BIGINT))
+                                .source(p.unnest(
+                                        ImmutableList.of(),
+                                        ImmutableList.of(new UnnestNode.Mapping(p.symbol("char_array"), ImmutableList.of(p.symbol("unnested_char")))),
+                                        Optional.empty(),
+                                        LEFT,
+                                        Optional.empty(),
+                                        p.project(
+                                                Assignments.of(p.symbol("char_array"), PlanBuilder.expression("regexp_extract_all(corr, '.')")),
+                                                p.values(ImmutableList.of(), ImmutableList.of(ImmutableList.of()))))))))
+                .matches(
+                        project(
+                                aggregation(
+                                        singleGroupingSet("corr", "unique", "char_array"),
+                                        ImmutableMap.of(Optional.of("max"), functionCall("max", ImmutableList.of("unnested_char"))),
+                                        ImmutableList.of(),
+                                        Optional.empty(),
+                                        SINGLE,
+                                        unnest(
+                                                ImmutableList.of("corr", "unique", "char_array"),
+                                                ImmutableList.of(unnestMapping("char_array", ImmutableList.of("unnested_char"))),
+                                                Optional.empty(),
+                                                LEFT,
+                                                Optional.empty(),
+                                                project(
+                                                        ImmutableMap.of("char_array", expression("regexp_extract_all(corr, '.')")),
+                                                        assignUniqueId("unique", values("corr")))))));
+    }
+
+    @Test
+    public void testMultipleNodesOverUnnestInSubquery()
```

Comment:
Thanks :) I just hope no one will ever need to edit this one :p

Reply:
Oh. I never thought that tests might get in the way of future progress. If they do, I'll be there to help.